src/palace/manager/api/admin/routes.py

```diff
@@ -219,6 +219,34 @@ def unsuppress(identifier_type, identifier):
     return app.manager.admin_work_controller.unsuppress(identifier_type, identifier)
 
 
+@library_route(
+    "/admin/works/<identifier_type>/<path:identifier>/suppress-for-library",
+    methods=["POST"],
+)
+@has_library
```

Comment:
I'm concerned about using the `has_library` decorator here, because if no library is provided, it will use the `default` library, which is arbitrary (usually, it's the first library created in a PM instance). In any instance that has more than one library, I don't think this is safe.

We could use the `allows_library` decorator, which will set `flask.request.library` to `None`, if no library is provided. In that case, we should probably handle the `library is None` case.